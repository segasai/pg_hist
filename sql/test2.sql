select * from pg_hist_w('select a,1 from test1',ARRAY[1], ARRAY[0],ARRAY[2]) 
    as (a int, b double precision);

select * from pg_hist_1d_w('select a,1 from test1',ARRAY[1], ARRAY[0],ARRAY[2]);
select * from pg_hist_1d_w('select b,1 from test1',ARRAY[1], ARRAY[0],ARRAY[2]);
select * from pg_hist_1d_w('select c,1 from test1',ARRAY[1], ARRAY[0],ARRAY[2]);
select * from pg_hist_1d_w('select d,1 from test1',ARRAY[1], ARRAY[0],ARRAY[2]);
select * from pg_hist_1d_w('select e,1 from test1',ARRAY[1], ARRAY[0],ARRAY[2]);

select * from pg_hist_1d_w('select a,1 from test1',ARRAY[100], ARRAY[-0.1],ARRAY[99.9]);
select * from pg_hist_1d_w('select b,1 from test1',ARRAY[100], ARRAY[-0.1],ARRAY[99.9]);
select * from pg_hist_1d_w('select c,1 from test1',ARRAY[100], ARRAY[-0.1],ARRAY[99.9]);
select * from pg_hist_1d_w('select d,1 from test1',ARRAY[100], ARRAY[-0.1],ARRAY[99.9]);
select * from pg_hist_1d_w('select e,1 from test1',ARRAY[100], ARRAY[-0.1],ARRAY[99.1]);

select * from pg_hist_2d_w('select a,b,1 from test1',ARRAY[100,100], ARRAY[-0.1,-0.1],ARRAY[99.9,99.9]);
select * from pg_hist_2d_w('select b,c,1 from test1',ARRAY[100,100], ARRAY[-0.1,-0.1],ARRAY[99.9,99.9]);
select * from pg_hist_2d_w('select c,d,1 from test1',ARRAY[100,100], ARRAY[-0.1,-0.1],ARRAY[99.9,99.9]);
select * from pg_hist_2d_w('select d,e,1 from test1',ARRAY[100,100], ARRAY[-0.1,-0.1],ARRAY[99.9,99.9]);
select * from pg_hist_2d_w('select e,a,1 from test1',ARRAY[100,100], ARRAY[-0.1,-0.1],ARRAY[99.9,99.9]);

select * from pg_hist_2d_w('select a,b*2,1 from test1',ARRAY[100,100], ARRAY[-0.1,-0.1],ARRAY[99.9,99.9]);
select * from pg_hist_2d_w('select b,c*2,1 from test1',ARRAY[100,100], ARRAY[-0.1,-0.1],ARRAY[99.9,99.9]);
select * from pg_hist_2d_w('select c,d*2,1 from test1',ARRAY[100,100], ARRAY[-0.1,-0.1],ARRAY[99.9,99.9]);
select * from pg_hist_2d_w('select d,e*2,1 from test1',ARRAY[100,100], ARRAY[-0.1,-0.1],ARRAY[99.9,99.9]);
select * from pg_hist_2d_w('select e,a*2,1 from test1',ARRAY[100,100], ARRAY[-0.1,-0.1],ARRAY[99.9,99.9]);

select * from pg_hist_1d_w('select 1.,1 from generate_series(1,100000);',ARRAY[1], ARRAY[0],ARRAY[2]);
select * from pg_hist_2d_w('select 1.,1.,1 from generate_series(1,100000);',ARRAY[1,1], ARRAY[0,0],ARRAY[2,2]);
select * from pg_hist_3d_w('select 1.,1.,1.,1 from generate_series(1,100000);',ARRAY[1,1,1], ARRAY[0,0,0],ARRAY[2,2,2]);

select * from pg_hist_w('select 1.,1.,1.,1.,1.,1 from generate_series(1,100000);',ARRAY[11,11,11,11,11], ARRAY[0,0,0,0,0],ARRAY[2,2,2,2,2]) 
    as (x1 int, x2 int, x3 int, x4 int, x5 int, hist double precision);
