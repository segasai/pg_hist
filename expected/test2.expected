 a | b 
---+---
 0 | 1
(1 row)

 x1 | hist 
----+------
  0 |    1
(1 row)

 x1 | hist 
----+------
  0 |    1
(1 row)

 x1 | hist 
----+------
  0 |    1
(1 row)

 x1 | hist 
----+------
  0 |    1
(1 row)

 x1 | hist 
----+------
  0 |    1
(1 row)

 x1 | hist 
----+------
  1 |    1
 11 |    1
(2 rows)

 x1 | hist 
----+------
  1 |    1
 11 |    1
(2 rows)

 x1 | hist 
----+------
  1 |    1
 11 |    1
(2 rows)

 x1 | hist 
----+------
  1 |    1
 11 |    1
(2 rows)

 x1 | hist 
----+------
  1 |    1
 11 |    1
(2 rows)

 x1 | x2 | hist 
----+----+------
  1 |  1 |    1
 11 | 11 |    1
(2 rows)

 x1 | x2 | hist 
----+----+------
  1 |  1 |    1
 11 | 11 |    1
(2 rows)

 x1 | x2 | hist 
----+----+------
  1 |  1 |    1
 11 | 11 |    1
(2 rows)

 x1 | x2 | hist 
----+----+------
  1 |  1 |    1
 11 | 11 |    1
(2 rows)

 x1 | x2 | hist 
----+----+------
  1 |  1 |    1
 11 | 11 |    1
(2 rows)

 x1 | x2 | hist 
----+----+------
  1 |  2 |    1
 11 | 22 |    1
(2 rows)

 x1 | x2 | hist 
----+----+------
  1 |  2 |    1
 11 | 22 |    1
(2 rows)

 x1 | x2 | hist 
----+----+------
  1 |  2 |    1
 11 | 22 |    1
(2 rows)

 x1 | x2 | hist 
----+----+------
  1 |  2 |    1
 11 | 22 |    1
(2 rows)

 x1 | x2 | hist 
----+----+------
  1 |  2 |    1
 11 | 22 |    1
(2 rows)

 x1 |  hist  
----+--------
  0 | 100000
(1 row)

 x1 | x2 |  hist  
----+----+--------
  0 |  0 | 100000
(1 row)

 x1 | x2 | x3 |  hist  
----+----+----+--------
  0 |  0 |  0 | 100000
(1 row)

 x1 | x2 | x3 | x4 | x5 |  hist  
----+----+----+----+----+--------
  5 |  5 |  5 |  5 |  5 | 100000
(1 row)

